### ISSUE TITLE
Ability to set Numba's `parallel` and `fastmath` options 

### ISSUE THREAD

**Issue by twiecki** [aesara-devs/aesara#409]
`numba.njit(parallel=True, fastmath=True)` can give great speed-ups. Would be nice to be able to run aesara with these flags.

**Comment by brandonwillard** [comment]
> `numba.njit(parallel=True, fastmath=True)` can give great speed-ups. Would be nice to be able to run aesara with these flags.

Yes, I've planned on setting better defaults for certain `Op`s, as well as enabling the caching option.

**Comment by brandonwillard** [comment]
To clarify this issue, we may want `aesara.config` options that allow users to specify whether or not the `fastmath` and/or `parallel` options are enabled for relevant `Op`s.

These options are&mdash;naturally&mdash;very `Op` dependent, so we need to first outline which ones are even applicable to which `Op`s.  The `fastmath` options seems like it could a simple global option, but the `parallel` setting is easily **not** something one would ever want set globally, since the result of parallelizing _all_ parallel-capable `Op`s could be quite bad on average.

In other words, the `parallel` option requires some node-level specificity to even make sense.  The most practical approach I can think of right now uses `.tag`: i.e. users would set something like `.tag.numba_njit_settings = {"parallel": True}` to manually interact with the `njit` options.

**Comment by brandonwillard** [comment]
These options were added in https://github.com/aesara-devs/aesara/pull/639.
